src/useCases/AdminManager.java

```diff
@@ -27,13 +29,18 @@ public AdminManager(Database database){
      * @param password password of new account.
      * @return Admin data if sign in successful, if username exists in the database, return null.
      */
-    public AdminData createAdmin(String username, String password){
-        Admin admin = new Admin(username, password);
-        if (adminDatabase.add(admin) != null) {
-            admin.setContactInfoId(newContactInDatabase());
-            return new AdminData(admin);
+    public AdminData createAdmin(String username, String password) {
+        if (Pattern.matches("^[a-zA-Z0-9]{6,}$", username) && Pattern.matches("^.{8,}$", password)) {
```

Comment:
It looks like you are repeating the same pattern across all UserManagers. Creating a helper method inside the UserManager abstract class that takes arguments `username` and `password` and returns a boolean would do you better.